lib/puppet/rest/client.rb

```diff
@@ -0,0 +1,97 @@
+require 'httpclient'
+
+require 'puppet'
+require 'puppet/rest/response'
+require 'puppet/rest/server_resolver'
+
+module Puppet::Rest
+  class Client
+    def self.default_client
```

Comment:
I would expect the `client` param of initialize to have a default of `HTTPClient.new(...)` and a class method called `default_client` to return a memoized instance of this class with the default params.

Reply:
I think my confusion may come from calling both this class and the third party implementation we're delegating to `client`.

Reply:
Yeah I think that might make more sense. I really only split out the default HTTPClient because it looked kind of nasty embedded in the wrapper class constructor, but I agree exposing that in a class method seems bad.

I'm still a little unclear whether we will want a memoized instance of this class or not. I think we might rather attach instances of it to objects that will control its lifecycle (e.g. the Configurer or the Application). Probably going to leave that piece for the next PR.